src/pgduckdb_background_worker.cpp

```diff
@@ -418,10 +416,24 @@ TriggerActivity(void) {
 
 	SpinLockAcquire(&BgwShmemStruct->lock);
 	BgwShmemStruct->activity_count++;
-	/* Force wake up the background worker */
-	if (BgwShmemStruct->bgw_latch) {
-		SetLatch(BgwShmemStruct->bgw_latch);
+	/* Iterate over all background workers and notify them */
+	const uint32_t numProcs = ProcGlobal->allProcCount;
+	for (uint32_t i = 0; i < numProcs; i++) {
+		PGPROC *proc = &ProcGlobal->allProcs[i];
+		if (!proc->isBackgroundWorker || proc->databaseId == InvalidOid) {
```

Comment:
As discussed in person, this loop is quite. Let's see if we can use a shared memory hash table instead. You can initialize it with `ShmemInitHash`. As maximum size we can probably use `max_worker_processes`, maybe with a little bit of extra buffer.